File: SimpleAccounting/app/src/main/java/com/emmanuelmess/simpleaccounting/dataloading/async/LoadPrevBalanceAsyncTask.java
package com.emmanuelmess.simpleaccounting.dataloading.async;

import android.os.AsyncTask;

import com.emmanuelmess.simpleaccounting.db.TableMonthlyBalance;

/**
 * @author Emmanuel
 *         on 4/12/2016, at 13:47.
 */

public class LoadPrevBalanceAsyncTask extends AsyncTask<Void, Void, Double> {

	private int month, year;
	private String currency;
	private TableMonthlyBalance tableMonthlyBalance;
	private AsyncFinishedListener<Double> listener;

	public LoadPrevBalanceAsyncTask(int m, int y, String c, TableMonthlyBalance db,
									AsyncFinishedListener<Double> l) {
		month = m;
		year = y;
		currency = c;
		tableMonthlyBalance = db;
		listener = l;
	}

	@Override
	protected void onPreExecute() {
		tableMonthlyBalance.getReadableDatabase();//updates the database, calls onUpgrade()
	}

	@Override
	protected Double doInBackground(Void... v) {
		return tableMonthlyBalance.getBalanceLastMonthWithData(month, year, currency);
	}

	@Override
	protected void onPostExecute(Double lastMonthData) {
		if(!isCancelled())
			listener.onAsyncFinished(lastMonthData);
	}
}

File: SimpleAccounting/app/src/main/java/com/emmanuelmess/simpleaccounting/dataloading/async/LoadPrevBalanceAsyncTask.kt
package com.emmanuelmess.simpleaccounting.dataloading.async

import android.os.AsyncTask

import com.emmanuelmess.simpleaccounting.db.TableMonthlyBalance

/**
 * @author Emmanuel
 * on 4/12/2016, at 13:47.
 */

class LoadPrevBalanceAsyncTask(
	private val month: Int,
	private val year: Int,
	private val currency: String,
	private val tableMonthlyBalance: TableMonthlyBalance,
	private val listener: AsyncFinishedListener<Double?>
) : AsyncTask<Void, Void, Double>() {

	@Suppress("UsePropertyAccessSyntax")
	override fun onPreExecute() {
		tableMonthlyBalance.getReadableDatabase()//updates the database, calls onUpgrade()
	}

	override fun doInBackground(vararg v: Void): Double? {
		return tableMonthlyBalance.getBalanceLastMonthWithData(month, year, currency)
	}

	override fun onPostExecute(lastMonthData: Double?) {
		if (!isCancelled) {
			listener.onAsyncFinished(lastMonthData)
		}
	}
}
